cfme/tests/cli/test_appliance_console_db_restore.py

```diff
@@ -0,0 +1,498 @@
+        pwd, TimedCommand(pwd, 360), '')
+    appl3.appliance_console.run_commands(command_set)
+    appl3.wait_for_evm_service()
+    appl3.wait_for_web_ui()
+    # Configure primary replication node
+    command_set = ('ap', '', '6', '1', '1', '', '', pwd, pwd, app0_ip, 'y',
+        TimedCommand('y', 60), '')
+    appl1.appliance_console.run_commands(command_set)
+    # Configure secondary replication node
+    command_set = ('ap', '', '6', '2', '1', '2', '', '', pwd, pwd, app0_ip, app1_ip, 'y',
+        TimedCommand('y', 60), '')
+    appl2.appliance_console.run_commands(command_set)
+    # Configure automatic failover on EVM appliance
+    command_set = ('ap', '', '9', TimedCommand('1', 30), '')
+    appl3.appliance_console.run_commands(command_set)
+
+    def is_ha_monitor_started(appliance):
+        return bool(appliance.ssh_client.run_command(
+            "grep {} /var/www/miq/vmdb/config/failover_databases.yml".format(app1_ip)).success)
+    wait_for(is_ha_monitor_started, func_args=[appl3], timeout=300, handle_exception=True)
+    # Add infra/cloud providers and create db backup
+    provider_app_crud(VMwareProvider, appl3).setup()
+    provider_app_crud(EC2Provider, appl3).setup()
+    appl1.db.backup()
+
+    return unconfigured_appliances
+
+
+def fetch_v2key(appl1, appl2):
+    # Fetch v2_key and database.yml from the first appliance
+    rand_v2_filename = "/tmp/v2_key_{}".format(fauxfactory.gen_alphanumeric())
+    rand_yml_filename = "/tmp/database_yml_{}".format(fauxfactory.gen_alphanumeric())
+    appl1.ssh_client.get_file("/var/www/miq/vmdb/certs/v2_key", rand_v2_filename)
+    appl2.ssh_client.put_file(rand_v2_filename, "/var/www/miq/vmdb/certs/v2_key")
+    appl1.ssh_client.get_file("/var/www/miq/vmdb/config/database.yml", rand_yml_filename)
+    appl2.ssh_client.put_file(rand_yml_filename, "/var/www/miq/vmdb/config/database.yml")
+
+
+def fetch_db_local(appl1, appl2):
+    # Fetch db from the first appliance
+    dump_filename = "/tmp/db_dump_{}".format(fauxfactory.gen_alphanumeric())
+    appl1.ssh_client.get_file("/tmp/evm_db.backup", dump_filename)
+    appl2.ssh_client.put_file(dump_filename, "/tmp/evm_db.backup")
+
+
+def setup_nfs_samba_backup(appl1):
+    # Fetch db from first appliance and push it to nfs/samba server
+    connect_kwargs = {
+        'hostname': cfme_data['network_share']['hostname'],
+        'username': credentials['ssh']['username'],
+        'password': credentials['depot_credentials']['password']
+    }
+    loc = cfme_data['network_share']['nfs_path']
+    nfs_smb = SSHClient(**connect_kwargs)
+    dump_filename = "/tmp/db_dump_{}".format(fauxfactory.gen_alphanumeric())
+    appl1.ssh_client.get_file("/tmp/evm_db.backup", dump_filename)
+    nfs_smb.put_file(dump_filename, "{}share.backup".format(loc))
+
+
+@pytest.mark.tier(2)
+@pytest.mark.uncollectif(lambda appliance: not appliance.is_downstream)
```

Comment:
Required: Please add a reason for uncollectif
ex. @pytest.mark.uncollectif(lambda appliance: not appliance.is_downstream, reason='Test only for downstream version of product')